cloudevents/pydantic/conversion.py

```diff
@@ -0,0 +1,68 @@
+#  Copyright 2018-Present The CloudEvents Authors
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+import typing
+
+from cloudevents.conversion import from_dict as _abstract_from_dict
+from cloudevents.conversion import from_http as _abstract_from_http
+from cloudevents.conversion import from_json as _abstract_from_json
+from cloudevents.conversion import to_binary, to_dict, to_json, to_structured  # noqa
+from cloudevents.pydantic.event import CloudEvent
+from cloudevents.sdk import types
+
+
+def from_http(
+    headers: typing.Dict[str, str],
+    data: typing.Union[str, bytes, None],
+    data_unmarshaller: typing.Optional[types.UnmarshallerType] = None,
+) -> CloudEvent:
+    """
+    Unwrap a CloudEvent (binary or structured) from an HTTP request.
+    :param headers: the HTTP headers
+    :param data: the HTTP request body. If set to None, "" or b'', the returned
+        event's data field will be set to None
+    :param data_unmarshaller: Callable function to map data to a python object
+        e.g. lambda x: x or lambda x: json.loads(x)
+    :type data_unmarshaller: types.UnmarshallerType
```

Comment:
This type looks excessive as we already have the typings. It also is confusing while there's no `Optional` here. I'd just remove this one.